src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/SPARQLStar2CypherTranslatorImpl.java

```diff
@@ -0,0 +1,474 @@
+                }
+            }
+            else if (o.isVariable()) {
+                if (configuration.isLabelIRI(p) && configuration.mapsToNode(s)) {
+                    return getNodeLabelVar(s, p, o, configuration, gen);
+                } else if (configuration.mapsToProperty(p) && configuration.mapsToNode(s)) {
+                    return getNodePropertyVar(s, p, o, configuration, gen);
+                } else if (configuration.mapsToEdgeLabel(p) && configuration.mapsToNode(s)) {
+                    return getNodeRelationshipVar(s, p, o, configuration, gen);
+                } else {
+                    return null;
+                }
+            }
+            else if (p.isVariable()) {
+                if (configuration.mapsToNode(s) && configuration.mapsToNode(o)) {
+                    return getNodeVarNode(s, p, o, configuration, gen);
+                } else if (configuration.mapsToNode(s) && o.isLiteral()) {
+                    return getNodeVarLiteral(s, p, o, configuration, gen);
+                } else if (configuration.mapsToNode(s) && configuration.mapsToLabel(o)) {
+                    return getNodeVarLabel(s, p, o, configuration, gen);
+                } else {
+                    return null;
+                }
+            }
+        } else if (nbOfVars == 2) {
+            if (s.isVariable() && o.isVariable()) {
+                if (configuration.isLabelIRI(p)) {
+                    return getVarLabelVar(s, p, o, configuration, gen);
+                } else if (configuration.mapsToEdgeLabel(p)) {
+                    return getVarRelationshipVar(s, p, o, configuration, gen);
+                } else if (configuration.mapsToProperty(p)) {
+                    return getVarPropertyVar(s, p, o, configuration, gen, certainNodes);
+                } else {
+                    return null;
+                }
+            } else if (s.isVariable() && p.isVariable()) {
+                if (configuration.mapsToLabel(o)) {
+                    return getVarVarLabel(s, p, o, configuration, gen);
+                } else if (configuration.mapsToNode(o)) {
+                    return getVarVarNode(s, p, o, configuration, gen);
+                } else if (o.isLiteral()) {
+                    return getVarVarLiteral(s, p, o, configuration, gen, certainNodes);
+                }
+            } else if(p.isVariable() && o.isVariable()) {
+                if (configuration.mapsToNode(s)) {
+                    return getNodeVarVar(s, p, o, configuration, gen, certainNodes);
+                }
+            }
+        } else {
+            return getVarVarVar(s, p, o, configuration, gen, certainNodes);
+        }
+        return null;
+    }
+
+    private static int getNumberOfVars(final Node s, final Node p, final Node o) {
+        int count = 0;
+        if (s.isVariable())
+            count++;
+        if (p.isVariable())
+            count++;
+        if (o.isVariable())
+            count++;
+        return count;
+    }
```

Comment:
No need for this helper function here. Instead, simply use `getNumberOfVarOccurrences` that we have in `QueryPatternUtils`.